client/client.go

```diff
@@ -641,8 +714,16 @@ type (
 		//  - serviceerror.Internal
 		//  - serviceerror.Unavailable
 		//  - serviceerror.NotFound
+		// Deprecated: Use [DescribeTaskQueueEnhanced]
 		DescribeTaskQueue(ctx context.Context, taskqueue string, taskqueueType enumspb.TaskQueueType) (*workflowservice.DescribeTaskQueueResponse, error)
 
+		// DescribeTaskQueueEnhanced  returns information about the target task queue, broken down by Build Id:
```

Comment:
What is the behaviour if `DescribeTaskQueueEnhanced` is called on an old server version that doesn't support the enhanced flag?

Reply:
Good question. It will return an empty `TaskQueueDescription{}`, I will add a comment...

Reply:
hm generally if you call an API that is not supported Go APIs and our APIs return an error.

Reply:
Let me think about this a bit more. I think we could infer that the server was "old" from the answer and return an error. The return values are not guaranteed to be correct, e.g., the request could filter by "nexus queue type" and that defaults to "workflow queue type" in old servers, so I think returning an error or nothing is the best path.

Also, I think we should not deprecate the old DescribeTaskQueue just yet. There are few users for that, mostly for getting pollers info, but we may want to give them a way to get that info in old servers without deprecation for a while.

Jumping ahead, for the cli, we probably should have a "legacy" flag to DescribeTaskQueue that defaults to the old DescribeTaskQueue

@cretz @Quinn-With-Two-Ns What do you think?

Reply:
If `DescribeTaskQueueResponse.versions_info` is empty/unset, is that enough to know that it's not supported server side? What situations would that value not be set when calling this?

> Jumping ahead, for the cli, we probably should have a "legacy" flag to DescribeTaskQueue that defaults to the old DescribeTaskQueue

Can discuss over there, but that sounds about right. We're going to break compatibility, but that may not be a concern at this stage. Worth discussion on that side.

Reply:
"If DescribeTaskQueueResponse.versions_info is empty/unset, is that enough to know that it's not supported server side? What situations would that value not be set when calling this?"

I think we also need to check that the server is returning info in the legacy proto fields "pollers" and "task-queue_status", and the map in "versions_info" is empty

Reply:
What situations would versions_info come back empty when we send a request for enhanced _except_ if it was an old server? Is it accepted that an enhanced info request would ever return this set as empty?

Reply:
I think that if select out the "Unversioned" queue in the queue selector, and there are no BuildDs used, that may give us an empty map (TBC). But the old server will populate the legacy fields with the "default" unversioned queue, so we could figure that out...

Reply:
Adding a `detectEnhancedNotSupported()` method, not deprecating vanilla `DescribeTaskQueue` just yet...